lib/elixir_bench/benchmarks/benchmarks.ex

```diff
@@ -51,6 +51,16 @@ defmodule ElixirBench.Benchmarks do
     Repo.fetch(where(Job, uuid: ^uuid))
   end
 
+  def fetch_running_job_by_reference(branch_name, commit_sha) do
+    query =
+      Job
+      |> last
+      |> Job.unfinished()
```

Comment:
why do we scope by unfinished? If it's the same sha for the same repo, I don't care if it's still running or not I don't think we should run it again unless a user explicitly requests this and we don't have this feature yet :)

Reply:
Yeah, in fact knowing if there is an existent job for a branch and commit should be enough, regardless if its running or finished. And I missed the repo part here.. it must be checked.

Reply:
make sense...